styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java

```diff
@@ -205,6 +215,64 @@ private void evaluateResourcesForDequeue(
         .map(resource -> ResourceWithInstance.create(resource, instanceState));
   }
 
+  @VisibleForTesting
+  void triggerBackfills(Collection<InstanceState> activeStates) {
+    final List<Backfill> backfills;
```

Comment:
Can we filter out backfills that are already completed at this preliminary stage? So we won't scan the cache for very old Workflows belonging to very old backfills which might not be in the cache anymore.